hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/tokens/TokenStore.java

```diff
@@ -83,22 +83,22 @@ default ResponseCodeEnum delete(TokenID id) {
     }
 
     default ResponseCodeEnum tryTokenChange(BalanceChange change) {
-        var validity = OK;
         var tokenId = resolve(change.tokenId());
         if (tokenId == MISSING_TOKEN) {
-            validity = INVALID_TOKEN_ID;
+            return INVALID_TOKEN_ID;
         }
+
         if (change.hasExpectedDecimals() && !matchesTokenDecimals(change.tokenId(), change.getExpectedDecimals())) {
-            validity = UNEXPECTED_TOKEN_DECIMALS;
+            return UNEXPECTED_TOKEN_DECIMALS;
```

Comment:
Original code return validity value at the end. Seems the same?

Reply:
Returning in the middle of the function behaves differently than returning at the end of the function. The problem here was that the code sets `validity = INVALID_TOKEN_ID`, _but does not immediately return that status_. Instead, the code goes on to compute `!matchesTokenDecimals(...)`, which expects an existing token ID, and therefore throws an exception.

Reply:
I didn't see !matchesTokenDecimals(...) below since it was not changed. Thanks for explaining.